### PR TITLE
[WIP] Add doc on protecting release branches

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Add some documentation or managing the release branches.

Related to PR #31426 

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME
`hacking/release-branches.rst`

##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

File: hacking/release-branches.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+During a release cycle, the release branch name ``stable-[version]`` will be protected to prevent direct pushes from collaborators and members who have commit access to the repository. Only the designated release managed can cherry pick commits from the upstream ``devel`` branch to the stable release branch. This restriction comes and goes during the release cycle, being enforced closer to release time and being removed after a release to allow contributors and members to cherry-pick commits that are appropriate for inclusion in the next dot release.
+
+The procedure for protecting branches on GitHub is as follows:
+
+**Note:** You must be a repository administrator in order to change these settings.
+
+1. Navigate the the main repository page.
+2. Click **Settings**
+3. In the left menue, click **Branches**.
```
Comment (mattclay) [inline hacking/release-branches.rst]: "menue" -> "menu"

File: hacking/release-branches.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+4. Under Protected Branches, select the branch you want to restrict.
```
Comment (mattclay) [inline hacking/release-branches.rst]: Under Protected Branches, click "Edit" for the branch you want to restrict.

File: hacking/release-branches.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+5. Select **Restrict who can push to this branch**.
+6. Search for and select the designated release manager.
```
Comment (mattclay) [inline hacking/release-branches.rst]: Add "Select **Include administrators**".

File: hacking/release-branches.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Mananging Release Branches
```
Comment (mattclay) [inline hacking/release-branches.rst]: "Mananging" -> "Managing"

**Review by mattclay — APPROVED** [review]
Looks good overall, just make the corrections noted.

**Comment by abadger** [comment]
I've copied this over to https://docs.google.com/document/d/1gWPtxNX4J39uIzwqQWLIsTZ1dY_AwEZzAd9bJ4XtZso/edit#heading=h.2wezayw9xsqz since we decided that all of these process notes would go into *Public* google docs.  Still need to add/update links in the production documentation to point to them.

**Comment by alexandrevicenzi** [comment]
@abadger Ansible Docs [GitHub Admins](https://docs.ansible.com/ansible/latest/community/github_admins.html) still has a broken link. Should it be updated to google docs that you referred in the previous comment?

**Pull request closed without merging**